pkg/apis/azuremulticlusteringress/v1alpha1/types.go

```diff
@@ -0,0 +1,36 @@
+// -------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// --------------------------------------------------------------------------------------------
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AzureMultiClusterIngress is the resource AGIC is watching in MultiCluster mode
+type AzureMultiClusterIngress struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec v1.IngressSpec `json:"spec"`
```

Comment:
If MCI Spec is Ingress, we won't be able to extend the spec

Reply:
sure, what extra properties should we add to spec in nested MultiIngressSpec ?